server.js

```diff
@@ -39,21 +39,48 @@ joinGame(isGenerator, resp) {
     game.setGenerator(id);
   }
   players.push(id);
-  resp.status(200).json(id);
+  resp.status(200).json({
+    id,
+    ready: players.length >= 2,
+    isGen: isGenerator}
+  );
 }
 
 startGame({ word, id }, resp) {
   if (!word || !id) {
-    return resp.status(400).json(`Word or id are missing. Word: ${word}. Id: ${id}`);
+    return resp.status(400).json(`Word is missing. Word: ${word}.`);
+  }
+  if (game.verifyGen(id)) {
+    return resp.status(401).json(`Not the generator. ID provided: ${id}.`);
   }
   game.setWordToGuess(word);
   if (players.length < 2) {
     return resp.status(200).json(`Received. Waiting on others to join...`);
   }
-  resp.status(200).json('Received. Game starting.')
+  resp.status(200).json({
+    display: game.displayRevealed(),
+    guesses: game.getGuessesLeft(),
+    isOver: game.isOver()
+  });
 }
 
-
+makeGuess({ guess, id }) {
+  if (!guess) {
+    return resp.status(400).json(`Guess is missing. Guess: ${guess}.`);
+  }
+  if (!game.verifyGen(id)) {
+    return resp.status(401).json(`Generator is not allowed to guess. ID provided: ${id}.`);
```

Comment:
So I think this will be part of the conditional rendering from React. If a player's state is set to isGen = true, they will not see an input field to make a guess, and thus will not have access to it's functionality.

Reply:
Awesome. I have no idea about client stuff, but when we link it up, we can remove redundancy later.

I do like the idea of making the game-host `hack-proof` though.... XD